textworld/generator/game.py

```diff
@@ -772,3 +779,125 @@ def update(self, action: Action) -> None:
         # Update all quest progressions given the last action and new state.
         for quest_progression in self.quest_progressions:
             quest_progression.update(action, self.state)
+
+
+class GameOptions:
+    """
+    Options for customizing the game generation.
+
+    Attributes:
+        nb_rooms:
+            Number of rooms in the game.
+        nb_objects:
+            Number of objects in the game.
+        quest_length:
+            Minimum number of actions the quest requires to be completed.
+        quest_breadth:
+            Control how nonlinear a quest can be (1: linear).
+        games_dir:
+            Path to the directory where the game will be saved.
+        force_recompile:
+            If `True`, recompile game even if it already exists.
+        file_type:
+            Type of the generated game file. Either .z8 (Z-Machine) or .ulx (Glulx).
+        seeds:
+            Seeds for the different generation processes.
+
+               * If `None`, seeds will be sampled from
+                 :py:data:`textworld.g_rng <textworld.utils.g_rng>`.
+               * If `int`, it acts as a seed for a random generator that will be
+                 used to sample the other seeds.
+               * If dict, the following keys can be set:
+
+                 * `'map'`: control the map generation;
+                 * `'objects'`: control the type of objects and their
+                   location;
+                 * `'quest'`: control the quest generation;
+                 * `'grammar'`: control the text generation.
+
+                 For any key missing, a random number gets assigned (sampled
+                 from :py:data:`textworld.g_rng <textworld.utils.g_rng>`).
+        chaining_options:
+            For customizing the quest generation (see
+            :py:class:`textworld.generator.ChainingOptions <textworld.generator.chaining.ChainingOptions>`
+            for the list of available options).
+        grammar_options:
+            For customizing the text generation (see
+            :py:class:`textworld.generator.GrammarOptions <textworld.generator.text_grammar.GrammarOptions>`
+            for the list of available options).
+    """
+
+    def __init__(self):
+        self.chaining_options = ChainingOptions()
```

Comment:
Maybe you can drop `_options` from these?  `options.chaining_options.whatever` is a bit redundant